modules/ROOT/pages/patterns/reference.adoc

```diff
@@ -0,0 +1,1324 @@
+
+[options="header",cols="1m,1m,1a"]
+|===
+| *Variant* | *Canonical form* | *Description*
+
+| {m,n} | {m,n} | Between m and n iterations.
+| + | {1,} | 1 or more iterations.
+| * | {0,} | 0 or more iterations.
+| \{n} | {n,n} | Exactly n iterations.
+| {m,} | {m,} | m or more iterations.
+| {,n} | {0,n} | Between 0 and n iterations.
+| {,} | {0,} | 0 or more iterations.
+
+|===
+
+Note that a xref:patterns/reference.adoc#quantified-path-patterns[quantified path pattern] with the quantifier `\{1}` is not equivalent to a fixed-length path pattern.
+Although the resulting quantified path pattern will match on the same paths the fixed-length path contained in it would without the quantifier, the presence of the quantifier means that all variables within the path pattern will be exposed as group variables.
+
+[[graph-patterns]]
+== Graph patterns
+
+A graph pattern is a comma separated list of one or more path patterns.
+It is the top level construct provided to `MATCH`.
+
+image::patterns_graph_pattern_reference.svg[width="500", role="middle"]
+
+[[graph-patterns-syntax]]
+=== Syntax
+
+[source, role=noheader]
+----
+graphPattern ::= 
+  pathPattern [ "," pathPattern ]* [ "WHERE" booleanExpression ]
+----
+
+[[graph-patterns-rules]]
+=== Rules
+
+The rules for xref:patterns/reference.adoc#path-patterns[path patterns] apply to each constituent path pattern of a graph pattern.
+
+[[graph-patterns-rules-variable-references]]
+==== Variable references
+
+Any node or relationship variable declared in a graph pattern can be referenced in a `WHERE` clause elsewhere in the graph pattern - unless it is inside a quantified path pattern not containing the variable.
+For example, this is allowed:
+
+[source, role=noheader]
+----
+(n)-->(m:A)-->(:B), (m)-[r WHERE r.p <> n.p]->(:C)
+----
+
+But this is not allowed:
+
+[source, role=noheader]
+----
+(n)-->(m:A)-->(:B), (m) (()-[r WHERE r.p <> n.p]->())+ (:C)
+----
+
+A variable can be referred to inside a quantified path pattern if it has already been bound in a previous `MATCH` clause.
+If a variable is declared inside a quantified path pattern, then it can be treated as a singleton only from within the quantified path pattern it was declared in.
+Outside of that quantified path pattern, it must be treated as a group variable.
+For example, this would be allowed:
+
+[source, role=noheader]
+----
+((n)-[r]->(m WHERE r.p = m.q))+
+----
+
+As would this:
+
+[source, role=noheader]
+----
+(n)-[r]->+(m WHERE all(rel in r WHERE rel.q > m.q))
+----
+
+But this would not be allowed:
+
+[source, role=noheader]
+----
+(n)-[r]->+(m WHERE r.p = m.q)
+----
+
+[[graph-patterns-rules-relationship-uniqueness]]
+==== Relationship uniqueness
+
+A relationship can only be traversed once in a given match for a graph pattern. The same restriction doesn't hold for nodes, which may be re-traversed any number of times in a match. 
+
+[[graph-patterns-rules-equijoin]]
+==== Equijoin
+
+_Equijoins were introduced in Neo4j 5.7._
```

Comment:
I guess we started calling it that in 5.7, and even then that's a bit of a stretch. I'd argue that it's been in Cypher forever(?).

Reply:
Ah, Ok. I misunderstood. Then I will delete the release version notice.